aiida/cmdline/params/options/config.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=cyclic-import
+"""
+.. py:module::config
+    :synopsis: Convenience class for configuration file option
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import yaml
+import click_config_file
+
+from .overridable import OverridableOption
+
+
+def yaml_config_file_provider(file_path, cmd_name):  # pylint: disable=unused-argument
+    """Read yaml config file."""
+    with open(file_path, 'r') as handle:
+        return yaml.load(handle)
+
+
+class ConfigFileOption(OverridableOption):
+    """
+    Wrapper around click_config_file.configuration_option that increases reusability.
+
+    Example::
+
+        CONFIG_FILE = ConfigFileOption('--config', help='A configuration file')
+
+        @click.command()
+        @click.option('computer_name')
+        @CONFIG_FILE(help='Configuration file for computer_setup')
+        def computer_setup(computer_name):
+            click.echo("Setting up computer {}".format(computername))
+
+        computer_setup --config config.yml
+
+    with config.yml::
+
+        ---
+        computer_name: computer1
+
+    """
+
+    def __init__(self, *args, **kwargs):  # pylint: disable=super-init-not-called
```

Comment:
I think it is better here to just create the correct `kwargs` and then call the super, instead of completely overriding.

Reply:
done